### PR TITLE
Add test cases for qcow2 volume with luks encryption

### PR DESCRIPTION
**Pull request by meinaLi**
Add test cases for qcow2 volume with luks encryption
1. Create qcow2 volume with luks encryption
2. Start vm with this volume

Signed-off-by: Meina Li <meili@redhat.com>

### CODE REVIEW

**Comment by meinaLi** [comment]
```
# avocado run --vt-type libvirt --vt-machine-type q35 virsh.vol_create.create.positive_test.fs_like_pool.vol_format.v_qcow2
JOB ID     : 9f259e8be570687e16fbf8ed4d46d57550ce650c
JOB LOG    : /root/avocado/job-results/job-2021-01-26T21.48-9f259e8/job.log
 (1/8) type_specific.io-github-autotest-libvirt.virsh.vol_create.create.positive_test.fs_like_pool.vol_format.v_qcow2.src_pool_type.dir.non_acl.luks_encryption: PASS (16.65 s)
 (2/8) type_specific.io-github-autotest-libvirt.virsh.vol_create.create.positive_test.fs_like_pool.vol_format.v_qcow2.src_pool_type.dir.non_acl.non_encryption: PASS (7.98 s)
 (3/8) type_specific.io-github-autotest-libvirt.virsh.vol_create.create.positive_test.fs_like_pool.vol_format.v_qcow2.src_pool_type.dir.acl_test.luks_encryption: PASS (32.20 s)
 (4/8) type_specific.io-github-autotest-libvirt.virsh.vol_create.create.positive_test.fs_like_pool.vol_format.v_qcow2.src_pool_type.dir.acl_test.non_encryption: PASS (17.14 s)
 (5/8) type_specific.io-github-autotest-libvirt.virsh.vol_create.create.positive_test.fs_like_pool.vol_format.v_qcow2.src_pool_type.fs.luks_encryption: PASS (46.60 s)
 (6/8) type_specific.io-github-autotest-libvirt.virsh.vol_create.create.positive_test.fs_like_pool.vol_format.v_qcow2.src_pool_type.fs.non_encryption: PASS (36.64 s)
 (7/8) type_specific.io-github-autotest-libvirt.virsh.vol_create.create.positive_test.fs_like_pool.vol_format.v_qcow2.src_pool_type.netfs.luks_encryption: PASS (23.03 s)
 (8/8) type_specific.io-github-autotest-libvirt.virsh.vol_create.create.positive_test.fs_like_pool.vol_format.v_qcow2.src_pool_type.netfs.non_encryption: PASS (14.06 s)
RESULTS    : PASS 8 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 197.96 s
```

File: libvirt/tests/src/virsh_cmd/volume/virsh_vol_create.py [hunk 1]
```diff
@@ -104,6 +112,43 @@ def run(test, params, env):
...
+                {"driver_type": vol_format,
+                 "source_encryption_dict": {"encryption": 'luks',
+                                            "secret": {"type": "passphrase",
+                                                       "uuid": uuid}}})
+        result = utlv.attach_additional_device(vm_name, target_dev,
+                                               disk_path, new_disk_dict)
+        if result.exit_status:
+            raise test.fail("Attach device %s failed." % target_dev)
+
+    def check_vm_start(params):
```
Comment (Yingshun) [inline libvirt/tests/src/virsh_cmd/volume/virsh_vol_create.py:139]: It looks like 'params' is unnecessary, how about removing it?

Comment (meinaLi) [reply libvirt/tests/src/virsh_cmd/volume/virsh_vol_create.py]: That's right and I've already updated it.

**Review by chunfuwen — APPROVED** [review]
LGTM

**Review by Yingshun — APPROVED** [review]

**Pull request merged**